crates/crypto/bls/src/pubkey.rs

```diff
@@ -37,3 +41,17 @@ impl PubKey {
         self.inner.iter().as_slice()
     }
 }
+
+impl FromStr for PubKey {
+    type Err = BLSError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let clean_str = s.strip_prefix("0x").unwrap_or(s);
+        let bytes = hex::decode(clean_str).map_err(|_| BLSError::InvalidHexString)?;
+        if bytes.len() != 48 {
+            return Err(BLSError::InvalidByteLength);
+        }
+        Ok(PubKey {
+            inner: FixedVector::from(bytes),
+        })
+    }
+}
```

Comment:
```suggestion

impl FromStr for PubKey {
    type Err = BLSError;
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        let clean_str = s.strip_prefix("0x").unwrap_or(s);
        let bytes = hex::decode(clean_str).map_err(|_| BLSError::InvalidHexString)?;
        
        if bytes.len() != 48 {
            return Err(BLSError::InvalidByteLength);
        }
        
        Ok(PubKey {
            inner: FixedVector::from(bytes),
        })
    }
}
```

its ok we don't need to remove all the sames. I think the spaces here help with readability, there is clearly 3 different chunks in this function
- decoding
- validation
- return